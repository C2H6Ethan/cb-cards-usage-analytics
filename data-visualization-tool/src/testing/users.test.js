import { create, act } from 'react-test-renderer';
import Users from '../components/users';
import axios from 'axios';

const users = [
	{ name: 'test user', date: new Date().toString() },
	{ name: 'test user 2', date: new Date().toString() },
];
const resp = { data: users };

const mockAxiosGet = jest.spyOn(axios, 'get');

mockAxiosGet.mockResolvedValue(resp);

it('renders correctly', async () => {
	// render the component
	act(() => {
		create(<Users />);
	});
});
