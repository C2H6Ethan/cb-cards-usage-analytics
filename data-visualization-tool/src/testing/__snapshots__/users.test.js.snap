// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    data-testid="usersTitle"
  >
    Users (
    0
    )
  </h2>
  <div
    className="table-responsive"
  >
    <table
      className="table table-bordered"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Username
          </th>
          <th>
            Date
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;

exports[`test 1`] = `
<div>
  <h2>
    Users (
    0
    )
  </h2>
  <div
    className="table-responsive"
  >
    <table
      className="table table-bordered"
    >
      <thead>
        <tr>
          <th>
            #
          </th>
          <th>
            Username
          </th>
          <th>
            Date
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
